mm/mm_heap/mm_free.c

```diff
@@ -86,6 +86,10 @@ void mm_free(FAR struct mm_heap_s *heap, FAR void *mem)
 
   kasan_poison(mem, mm_malloc_size(mem));
 
+  /* Must not in IRQ */
+
+  DEBUGASSERT(!up_interrupt_context());
```

Comment:
In the case we are in an interrupt context, this will nullify the changes we have in this PR for `mm_takesemaphore`.   
This check seems to be already implemented by the usual `DEBUGVERIFY(mm_takesemaphore)`.

Reply:
So you mean remove this check in mm_free() ?
But in order to align the code with mm_malloc(), then suggest add this check

Reply:
I was actually commenting about this line for all other files.  But it looks like I was wrong, I was thinking about `free` while the idea of this change is to use `mm_checkcorruption` in an interrupt context.

Reply:
Add the purpose to Summary